File: backend/app/models/assessment.py
from typing import List
from pydantic import BaseModel

from .client import Client
from .score import Scores
from .note import ClinicalNote

class AssessmentInfo(BaseModel):
    type: str
    full_name: str
    date_administered: str
    examiner: str
    status: str

class Assessment(BaseModel):
    id: str
    client: Client
    assessment: AssessmentInfo
    scores: Scores
    clinical_notes: List[ClinicalNote]

class AssessmentResponse(BaseModel):
    id: str
    client_name: str
    assessment_type: str
    date_administered: str
    status: str

File: frontend/src/types/assessment.ts
import type { Client } from './client';
import type { ClinicalNote } from './note';
import type { Scores } from './score';

export interface AssessmentInfo {
  type: string;
  fullName: string;
  dateAdministered: string;
  examiner: string;
  status: string;
}

export interface Assessment {
  id: string;
  client: Client;
  assessment: AssessmentInfo;
  scores: Scores;
  clinicalNotes: ClinicalNote[];
}

export interface AssessmentListResponse {
  id: string;
  clientName: string;
  assessmentType: string;
  dateAdministered: string;
  status: string;
}
